### PR TITLE
update porter to v1.0.0-rc.1

### PR DESCRIPTION
**Pull request by VinozzZ**
- update porter to v1.0.0-rc.1 and set it as the default porter-agent version
- use kubernetes-plugin v1.0.0-rc.1 in integration test
- update getporter/magefiles to v0.3.2
After this PR is merged, we will do release v0.7.0 for the porter-operator

### CODE REVIEW

File: magefiles/magefile.go [hunk 1]
```diff
@@ -161,7 +157,7 @@ func getPlugins() error {
 		feed    string
 		version string
 	}{
-		{name: "kubernetes", feed: "https://cdn.porter.sh/plugins/atom.xml", version: "latest"},
```
Comment (VinozzZ) [inline magefiles/magefile.go:164]: It doesn't look like we are publishing to the latest tag anymore

**Comment by VinozzZ** [comment]
This PR needs to wait for the ghcr.io/getporter/dev/porter-agent-kubernetes:v1.0.0-rc.1 to be published first

File: magefiles/magefile.go (hunk 1 above)
Comment (carolynvs) [reply magefiles/magefile.go:164]: When you run `porter plugin install` and use the latest version, it downloads the atom.xml file and selects the highest version of that plugin, excluding pre-releases.

**Review by carolynvs — APPROVED** [review]
LGTM!

**Pull request merged**